src/constructor/defaults.nim

```diff
@@ -61,26 +62,21 @@ macro defaults*(tdef: untyped): untyped =
         constrParams.add newColonExpr(ident.NimNode.basename, identDef.val)
       identDef.val = emptyNode
 
-  let objCstr = nnkObjConstr.newTree(constrParams)
-  var newProc = newProc(procName, params, objCStr)
-  defaultTable[result.repr.replace("*")] = newProc
+  let objCstr = nnkObjConstr.newTree(constrParams) # Object constructor: Obj(field1: val1, field2: val2)
 
-macro implDefaults*(t: typedesc[typed], genFlags: static set[DefaultFlag]): untyped =
-  ## Implements the default intializing procedure
-  ## Flags can be passed to change behaviour.
-  ## Refer to DefaultFlag to see behaviour of those flags.
-  result = defaultTable[t.getImpl.repr.replace("*")].copyNimTree
-  let routine = routineNode(result)
-  if defTypeConstr in genFlags:
-    routine.insertIdentDef 0, identDefTyp("_", routine.returnType.makeTypeDesc())
-    let name =
-      if result[0].strVal.startsWith("new"):
-        ident"new"
-      else:
-        ident"init"
-    result[0] = name
-  if defExported in genFlags:
-    result[0] = result[0].postfix("*")
+  let newProc = newProc(if defExported in genFlags: procName.postfix("*") else: procName, params, objCStr) # Initialization procedure
```

Comment:
I'd prefer these if expressions to be on their own lines. Inline if expressions are terrible for readability.

Reply:
Done :]